src/mlpack/methods/neat/genome.hpp

```diff
@@ -0,0 +1,319 @@
+/**
+ * @file genome.hpp
+ * @author Rahul Ganesh Prabhu
+ *
+ * Definition of the Genome class which represents a genome in the population.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_NEAT_GENOME_HPP
+#define MLPACK_METHODS_NEAT_GENOME_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "gene.hpp"
+
+namespace mlpack{
+namespace neat /** NeuroEvolution of Augmenting Topologies */ {
+
+/**
+ * A class representation of a genome in the population. The genome is a
+ * collection of connection genes which can be decoded into a neural
+ * network and be used to evaluate an input.
+ * 
+ * It also stores the nodes in the form of IDs:
+ * Bias node : ID = 0
+ * Input nodes : ID = 1 - inputNodeCount
+ * Output nodes : ID = inputNodeCount + 1 - inputNodeCount + outputNodeCount
+ * Hidden nodes : ID > inputNodeCount + outputNodeCount
+ * 
+ * The genome can undergo three types of structural mutations:
+ * 1. A connection can be split into two, creating a new node.
+ * 2. A new connection can be created between two nodes.
+ * 3. A connection can be deleted.
+ * 
+ * @tparam ActivationFunction The activation function to be used.
+ */
+template <class ActivationFunction = ann::HardSigmoidFunction>
+class Genome
```

Comment:
Since we use ``Genome<>`` a lot I think it would make sense from the style perspective, to use ``Genome`` instead, so we could write:

```
template <class ActivationFunction = ann::HardSigmoidFunction>
class GenomeType
{
};

template <class ActivationFunction = HardSigmoidFunction>
using Genome = GenomeType< ActivationFunction>;
```

Let me know what you think.